core/base/inc/TString.h

```diff
@@ -132,15 +139,27 @@ friend TString operator+(const TString &s1, const TString &s2);
 friend TString operator+(const TString &s,  const char *cs);
 friend TString operator+(const char *cs, const TString &s);
 friend TString operator+(const TString &s, char c);
-friend TString operator+(const TString &s, Long_t i);
-friend TString operator+(const TString &s, ULong_t i);
-friend TString operator+(const TString &s, Long64_t i);
-friend TString operator+(const TString &s, ULong64_t i);
 friend TString operator+(char c, const TString &s);
-friend TString operator+(Long_t i, const TString &s);
-friend TString operator+(ULong_t i, const TString &s);
-friend TString operator+(Long64_t i, const TString &s);
-friend TString operator+(ULong64_t i, const TString &s);
+
+template<class T>
+friend typename std::enable_if<is_signed_numeral<T>::value,TString>::type
```

Comment:
In ROOT we decided to use a default template parameter to specify SFINAE conditions, wherever possible, i.e. `template<class T, class = std::enable_if<is_signed_numeral<T>::value>::type>`. Would that work in this case?

Reply:
No, it would not. See my comment below.